pkg/asset/installconfig/installconfig.go

```diff
@@ -75,23 +75,7 @@ func (a *InstallConfig) Generate(parents asset.Parents) error {
 	a.Config.GCP = platform.GCP
 	a.Config.BareMetal = platform.BareMetal
 
-	if err := a.setDefaults(); err != nil {
-		return errors.Wrap(err, "failed to set defaults for install config")
-	}
-
-	if err := validation.ValidateInstallConfig(a.Config, openstackvalidation.NewValidValuesFetcher()).ToAggregate(); err != nil {
-		return errors.Wrap(err, "invalid install config")
-	}
-
-	data, err := yaml.Marshal(a.Config)
```

Comment:
would like to keep this part in the asset itself.

and `finish` could take `*InstallConfig` for better api.

Reply:
> would like to keep this part in the asset itself.

We currently marshal [in `Generate`][1] and [in `Load`][2] (presumably to capture any default injection so it doesn't need to get re-injected next time, it was added [here][3]).  So I think it makes sense to have it in the shared `finish` helper, but can push it back up into the doubled `Generate` / `Load` logic if you prefer.  Thoughts?

> and `finish` could take `*InstallConfig` for better api.

You want `func finish(a *InstallConfig, filename string) error` instead of `func (a *InstallConfig) finish(filename string) error`?  I don't see a benefit to a function over my current method, but I can reroll to use a function if you like.  Just let me know that I'm reading your comment correctly.

[1]: https://github.com/openshift/installer/blob/1405312dceec36e33bbaf0b6c51028a6db19c338/pkg/asset/installconfig/installconfig.go#L86
[2]: https://github.com/openshift/installer/blob/1405312dceec36e33bbaf0b6c51028a6db19c338/pkg/asset/installconfig/installconfig.go#L139
[3]: https://github.com/openshift/installer/commit/ec92355be7d70b076439acd22ef0e4703826341a#diff-74e3d89f2a13e465091f905f65cfc646R126

Reply:
don't care enough. It's fine as is.